src/shell/shell.zig

```diff
@@ -1778,6 +1800,9 @@ pub fn NewLexer(comptime encoding: StringEncoding) type {
                         switch (char) {
                             '0'...'9' => {
                                 _ = self.eat();
+                                if (count >= 32) {
+                                    return null;
+                                }
                                 buf[count] = @intCast(char);
```

Comment:
this wasnt changed in this pr but why is this `@intCast`. i dont actually know the types here but why can this cast fail?